src/Microsoft.ML.StandardTrainers/Standard/ModelStatistics.cs

```diff
@@ -16,249 +16,343 @@
 using Microsoft.ML.Trainers;
 
 // This is for deserialization from a model repository.
-[assembly: LoadableClass(typeof(LinearModelStatistics), null, typeof(SignatureLoadModel),
+[assembly: LoadableClass(typeof(ModelStatisticsBase), typeof(LinearModelParameterStatistics), null, typeof(SignatureLoadModel),
     "Linear Model Statistics",
-    LinearModelStatistics.LoaderSignature)]
+    LinearModelParameterStatistics.LoaderSignature)]
+
+// This is for deserialization from a model repository.
+[assembly: LoadableClass(typeof(ModelStatisticsBase), typeof(ModelStatisticsBase), null, typeof(SignatureLoadModel),
+    "Model Statistics",
+   ModelStatisticsBase.LoaderSignature)]
 
 namespace Microsoft.ML.Trainers
 {
     /// <summary>
     /// Represents a coefficient statistics object.
     /// </summary>
-    public readonly struct CoefficientStatistics
+    public sealed class CoefficientStatistics
     {
-        public readonly string Name;
         public readonly float Estimate;
         public readonly float StandardError;
         public readonly float ZScore;
         public readonly float PValue;
 
-        internal CoefficientStatistics(string name, float estimate, float stdError, float zScore, float pValue)
+        internal CoefficientStatistics(float estimate, float stdError, float zScore, float pValue)
         {
-            Contracts.AssertNonEmpty(name);
-            Name = name;
             Estimate = estimate;
             StandardError = stdError;
             ZScore = zScore;
             PValue = pValue;
         }
     }
 
-    // REVIEW: Make this class a loadable class and implement ICanSaveModel.
     // REVIEW: Reconcile with the stats in OLS learner.
     /// <summary>
     /// The statistics for linear predictor.
     /// </summary>
-    public sealed class LinearModelStatistics : ICanSaveModel
+    public class ModelStatisticsBase : ICanSaveModel
     {
-        internal const string LoaderSignature = "LinearModelStats";
-
-        private static VersionInfo GetVersionInfo()
-        {
-            return new VersionInfo(
-                modelSignature: "LMODSTAT",
-                verWrittenCur: 0x00010001, // Initial
-                verReadableCur: 0x00010001,
-                verWeCanReadBack: 0x00010001,
-                loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(LinearModelStatistics).Assembly.FullName);
-        }
-
-        private readonly IHostEnvironment _env;
+        protected IHostEnvironment Env;
```

Comment:
>protected [](start = 8, length = 9)

Protected members are still part of the public surface. `private protected` please.